src/session/content/chat_action_bar.rs

```diff
@@ -282,11 +287,183 @@ impl ChatActionBar {
+                        .chain_property::<Supergroup>("is-channel");
+                    let status_expression =
+                        gtk::ConstantExpression::new(data).chain_property::<Supergroup>("status");
+                    gtk::ClosureExpression::new::<bool, _, _>(
+                        &[
+                            permissions_expression.clone(),
+                            status_expression,
+                            channel_expression,
+                        ],
+                        closure!(|_: Chat,
+                                  permissions: BoxedChatPermissions,
+                                  status: BoxedChatMemberStatus,
+                                  is_channel: bool| {
+                            if is_channel {
+                                matches!(
+                                    status.0,
+                                    ChatMemberStatus::Creator(_)
+                                        | ChatMemberStatus::Administrator(_)
+                                )
+                            } else {
+                                match status.0 {
+                                    ChatMemberStatus::Restricted(data)
+                                        if !data.permissions.can_send_messages =>
+                                    {
+                                        data.permissions.can_send_messages
+                                    }
+                                    ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                    _ => permissions.0.can_send_messages,
+                                }
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+                ChatType::BasicGroup(data) => {
+                    let status_expression =
+                        gtk::ConstantExpression::new(data).chain_property::<BasicGroup>("status");
+                    gtk::ClosureExpression::new::<bool, _, _>(
+                        &[permissions_expression.clone(), status_expression],
+                        closure!(|_: Chat,
+                                  permissions: BoxedChatPermissions,
+                                  status: BoxedChatMemberStatus| {
+                            match status.0 {
+                                ChatMemberStatus::Restricted(data)
+                                    if !data.permissions.can_send_messages =>
+                                {
+                                    data.permissions.can_send_messages
+                                }
+                                ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                _ => permissions.0.can_send_messages,
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+            };
+
+            let message_bar_binding =
```

Comment:
I think the name `message_bar_binding` binding is not very accurate. I would call it `message_bar_visibility_binding`.